vote/vote_collector.go

```diff
@@ -14,18 +15,20 @@ import (
 )
 
 type VoteCollector struct {
-	config       *config.Config
-	executor     *executor.Executor
-	mtx          sync.RWMutex
-	dataProvider DataProvider
+	config        *config.Config
+	executor      *executor.Executor
+	mtx           sync.RWMutex
+	dataProvider  DataProvider
+	metricService *metrics.MetricService
 }
 
-func NewVoteCollector(cfg *config.Config, executor *executor.Executor, collectorDataProvider DataProvider) *VoteCollector {
+func NewVoteCollector(cfg *config.Config, executor *executor.Executor, collectorDataProvider DataProvider, metricService *metrics.MetricService) *VoteCollector {
```

Comment:
Can we also add some metrics for vote collection?

Reply:
added